File: Abstract Data Types/Symbol.py
# Python has no direct Symbol equivalent; use a private-like attribute name.


class User:
    def __init__(self, name, user_id):
        self.name = name
        self._id = user_id

    def get_id(self):
        return self._id


u = User("Marco", 42)
print(list(vars(u).keys()))
print({"name": u.name})
print(u.get_id())

File: Abstract Data Types/Symbol.js
const _id = Symbol("id");

class User {
  constructor(name, id) {
    this.name = name;
    this[_id] = id;   // Symbol key — not in for...in or Object.keys()
  }

  getId() { return this[_id]; }
}

const u = new User("Marco", 42);
console.log(Object.keys(u));        // ["name"] — id is hidden
console.log(JSON.stringify(u));     // {"name":"Marco"} — Symbol omitted
console.log(u.getId());             // 42